services-camel/src/test/java/edu/unc/lib/dl/services/camel/fulltext/FulltextRouterTest.java

```diff
@@ -60,14 +74,60 @@ protected AbstractApplicationContext createApplicationContext() {
         return new ClassPathXmlApplicationContext("/service-context.xml", "/fulltext-context.xml");
     }
 
+    @After
+    public void cleanup() throws IOException {
+        FileUtils.deleteDirectory(new File("target/34"));
+    }
+
+    @Test
+    public void testFullTextExtractionFilterValidMimeTypeNoForceNoExistingFile() throws Exception {
+        getMockEndpoint("mock:direct:fulltext.extraction").expectedMessageCount(1);
+        createContext(ENHANCEMENT_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent());
+        assertMockEndpointsSatisfied();
+    }
+
     @Test
-    public void testFullTextExtractionFilterValidMimeType() throws Exception {
+    public void testFullTextExtractionFilterValidMimeTypeForceNoExistingFile() throws Exception {
         getMockEndpoint("mock:direct:fulltext.extraction").expectedMessageCount(1);
         createContext(ENHANCEMENT_ROUTE);
 
+        Map<String, Object> headers = createEvent();
+        headers.put("force", "true");
+        template.sendBodyAndHeaders("", headers);
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void testFullTextExtractionFilterValidMimeTypeNoForceExistingFile() throws Exception {
+        String derivativePath = idToPath(fileID, HASHED_PATH_DEPTH, HASHED_PATH_SIZE);
+        File existingFile = new File("target/" + derivativePath + "/" + fileID + ".txt");
+        FileUtils.writeStringToFile(existingFile, "extracted text", "utf-8");
+
+        getMockEndpoint("mock:direct:fulltext.extraction").expectedMessageCount(0);
+        createContext(ENHANCEMENT_ROUTE);
+
+        Map<String, Object> headers = createEvent();
         template.sendBodyAndHeaders("", createEvent());
+        assertMockEndpointsSatisfied();
+        existingFile.delete();
+    }
+
+    @Test
+    public void testFullTextExtractionFilterValidMimeTypeForceExistingFile() throws Exception {
+        String derivativePath = idToPath(fileID, HASHED_PATH_DEPTH, HASHED_PATH_SIZE);
+        File existingFile = new File("target/" + derivativePath + "/" + fileID + ".txt");
+        FileUtils.writeStringToFile(existingFile, "extracted text", "utf-8");
+
+        getMockEndpoint("mock:direct:fulltext.extraction").expectedMessageCount(1);
+        createContext(ENHANCEMENT_ROUTE);
 
+        Map<String, Object> headers = createEvent();
+        headers.put("force", "true");
+        template.sendBodyAndHeaders("", headers);
         assertMockEndpointsSatisfied();
+        existingFile.delete();
```

Comment:
You can probably skip these delete steps since you have the `@After` method which recursively deletes the dir. If you didn't, this delete could be problematic since if the test failed it wouldn't delete the file.